core/internal/cltest/cltest.go

```diff
@@ -467,6 +479,10 @@ func NewApplicationWithConfig(t testing.TB, tc *TestConfig, flagsAndDeps ...inte
 
 	app.Store.Config = tc.Config
 
+	if !useRealExternalInitiatorManager {
+		app.ExternalInitiatorManager = externalInitiatorManager
+	}
+
```

Comment:
This feels weird. Can we not do it the same way we did the others? i.e. use the real one unless a mock is passed in

Reply:
We can't, because it has to be instantiated with a gorm.DB which is buried within store/orm. This was my gripe to you on Slack the other day. I agree, it's weird and shitty, but I have a plan to fix this across the whole testing codebase and application.go

Reply:
Urgh yeah, I hate that